java.lang.NullPointerException on package upgrade

```
type: crash
osVersion: google/raven/raven:14/AP2A.240805.005.F1/2024080600:user/release-keys
package: com.android.packageinstaller:34
process: com.android.packageinstaller
processUptime: 162 + 328 ms

java.lang.RuntimeException: Unable to start activity ComponentInfo{com.android.packageinstaller/com.android.packageinstaller.PackageInstallerActivity}: java.lang.NullPointerException: Attempt to get length of null array
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4004)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4184)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:114)
	at android.app.servertransaction.TransactionExecutor.executeNonLifecycleItem(TransactionExecutor.java:231)
	at android.app.servertransaction.TransactionExecutor.executeTransactionItems(TransactionExecutor.java:152)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:93)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2602)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loopOnce(Looper.java:232)
	at android.os.Looper.loop(Looper.java:317)
	at android.app.ActivityThread.main(ActivityThread.java:8623)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:580)
	at com.android.internal.os.ExecInit.main(ExecInit.java:50)
	at com.android.internal.os.RuntimeInit.nativeFinishInit(Native Method)
	at com.android.internal.os.RuntimeInit.main(RuntimeInit.java:369)
Caused by: java.lang.NullPointerException: Attempt to get length of null array
	at com.android.packageinstaller.PackageUtil.getPackageInfo(PackageUtil.java:79)
	at com.android.packageinstaller.PackageInstallerActivity.processPackageUri(PackageInstallerActivity.java:664)
	at com.android.packageinstaller.PackageInstallerActivity.processAppSnippet(PackageInstallerActivity.java:711)
	at com.android.packageinstaller.PackageInstallerActivity.onCreate(PackageInstallerActivity.java:468)
	at android.app.Activity.performCreate(Activity.java:9013)
	at android.app.Activity.performCreate(Activity.java:8991)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1531)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3986)
	... 15 more
```
Happened after upgrading apps using Aurora Store. Not sure which app caused it as they all seemed to install successfully.

--- a/packages/PackageInstaller/src/com/android/packageinstaller/PackageUtil.java
+++ b/packages/PackageInstaller/src/com/android/packageinstaller/PackageUtil.java
@@ -76,7 +76,8 @@ public class PackageUtil {
         String filePath = sourceFile.getAbsolutePath();
         if (filePath.endsWith(SPLIT_BASE_APK_END_WITH)) {
             File dir = sourceFile.getParentFile();
-            if (dir.listFiles().length > 1) {
+            File[] dirContents = dir.listFiles();
+            if (dirContents != null && dirContents.length > 1) {
                 // split apks, use file directory to get archive info
                 filePath = dir.getPath();
             }

